### PR TITLE
Ensure build and run image have the same stack as the stack.spec.id

### LINKED ISSUE #272: Ensure stack resource validates stack id across resource
Currently, the stack resource determines the `id` of the stack from the id field in the spec of the resource and does not verify if it matches the id specified in the `io.buildpacks.stack.id` of the build and the run image. Additionally, the build and run image can have different stack id's which should be a "broken" stack, but a stack still gets built. `kpack` should have stricter validations on stack inputs

### PR DESCRIPTION
**Pull request by ashwin-venkatesh**
- Make the stack resource not ready if there is a mismatch in the stack mentioned by the images and the spec
- Update openAPI module version

Closes #272

### CODE REVIEW

File: pkg/reconciler/v1alpha1/stack/stack.go [hunk 1]
```diff
@@ -123,17 +126,53 @@ func (c *Reconciler) getStackImages(stackSpec expv1alpha1.StackSpec) (string, st
...
+
+func getStackLabel(image ggcrv1.Image) (string, error) {
+	config, err := image.ConfigFile()
+	if err != nil {
+		return "", err
+	}
+	labels := config.Config.Labels
+	id, ok := labels[StackLabel]
+	if !ok {
+		return "", errors.New("invalid stack image")
```
Comment (ashwin-venkatesh) [inline pkg/reconciler/v1alpha1/stack/stack.go]: will improve this to be specific about which image is the invalid one

File: pkg/reconciler/v1alpha1/stack/stack.go [hunk 2]
```diff
@@ -123,17 +126,53 @@ func (c *Reconciler) getStackImages(stackSpec expv1alpha1.StackSpec) (string, st
...
+	}
+
+	if (buildStack == stackId) && (runStack == stackId) {
+		return nil
+	} else {
+		return errors.New(fmt.Sprintf("Invalid stack images. Expected stack: %s, Build Image stack: %s, Run Image stack: %s", stackId, buildStack, runStack))
+	}
+}
+
+func getStackLabel(image ggcrv1.Image) (string, error) {
```
Comment (matthewmcnew) [inline pkg/reconciler/v1alpha1/stack/stack.go]: This could just use `imagehelpers.GetStringLabel`

File: pkg/reconciler/v1alpha1/stack/stack.go [hunk 3]
```diff
@@ -123,17 +126,53 @@ func (c *Reconciler) getStackImages(stackSpec expv1alpha1.StackSpec) (string, st
...
+
+	runStack, err := getStackLabel(runImage)
+	if err != nil {
+		return err
+	}
+
+	if (buildStack == stackId) && (runStack == stackId) {
+		return nil
+	} else {
+		return errors.New(fmt.Sprintf("Invalid stack images. Expected stack: %s, Build Image stack: %s, Run Image stack: %s", stackId, buildStack, runStack))
```
Comment (matthewmcnew) [inline pkg/reconciler/v1alpha1/stack/stack.go]: Could you use `errors.Errorf` instead of `errors.New(fmt.Sprintf` ?

Comment (matthewmcnew) [reply pkg/reconciler/v1alpha1/stack/stack.go]: Also, super minor but, could we avoid capitalizing the error string? 
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Pull request merged**